libmcount/dynamic.c

```diff
@@ -302,6 +311,15 @@ static int find_dynamic_module(struct dl_phdr_info *info, size_t sz, void *data)
 	struct uftrace_mmap *map;
 	bool is_executable = mcount_is_main_executable(info->dlpi_name, sym_info->filename);
 
+	if (is_executable) {
+		if (main_loaded)
+			return !fmd->needs_modules;
+	}
+	else if (!fmd->needs_modules)
+		return main_loaded;
+	else if (modules_loaded)
+		return 0;
```

Comment:
Shouldn't it return 1?  Anyway, I think we don't call this function if `modules_loaded` set already.

Reply:
This statement will be executed if 
the current shared object is NOT the executable AND we need modules AND modules are loaded
I had this because of the original way I had the logic working, at this point it is irrelevant espacially if we don't check the return status of dl_iterate. However, I would not like tp speculate and play it safe so I decided not to return an error code here.